lib/al/Library/Math/MathUtil.cpp

```diff
@@ -1,5 +1,39 @@
 #include <al/Library/Math/MathUtil.h>
 
 namespace al {
-
+    void alongVectorNormalH(sead::Vector3f* out, const sead::Vector3f& param_2, const sead::Vector3f& param_3, const sead::Vector3f& param_4) {
+        auto vec2 = param_2;
+        auto vec3 = param_3;
+        auto vec4 = param_4;
+        sead::Vector3f cross;
+        sead::Vector3CalcCommon<f32>::cross(cross, vec3, vec4);
+        float mag = cross.x * cross.x + cross.y * cross.y + cross.z * cross.z;
+        if (mag >= 1e-06) {
+            f32 dot = sead::Vector3CalcCommon<f32>::dot(vec2, vec3);
```

Comment:
```suggestion
            f32 dot = vec2.dot(vec3);
```
similar below

Reply:
i removed this little half-assed decomp because it was gross and also didn't really need it right now lol